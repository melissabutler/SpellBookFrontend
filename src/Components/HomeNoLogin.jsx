import React from "react";
import { Link } from "react-router-dom";

const HomeNoLogin = () => {
    return (
        <div className="Home-nologin">
            <h1> Not logged in</h1>
            <button className="Home-button"><Link to="/login">Log In</Link></button>
            <button className="Home-button"><Link to="/signup">Sign Up</Link></button>
       
        </div>
    )   
}

export default HomeNoLogin;