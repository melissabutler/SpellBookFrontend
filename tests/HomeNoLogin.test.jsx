import React, { useContext } from "react";
import { describe, it, expect, vi } from "vitest";
import { render } from "@testing-library/react";
import HomeNoLogin from "../src/Components/HomeNoLogin";

describe(" HomeNoLogin", function() {

    vi.mock('react', async (importOriginal) => {
        const react = await importOriginal('react');
        return {
            ...react,
            useContext: () => ({ currentUser: "null" }),
            useRef: () => { target: null }
    }})

    vi.mock('react-router-dom', async () => {
        const mod = await vi.importActual('react-router-dom');
        return {
            ...mod,
            Link: () => ""
        }
    })


    it("renders without crashing", async function() {
        render(<HomeNoLogin />)
    })

    it("matches snapshot", function() {
        const { asFragment } = render(<HomeNoLogin />);
        expect(asFragment()).toMatchSnapshot();
    })



})
