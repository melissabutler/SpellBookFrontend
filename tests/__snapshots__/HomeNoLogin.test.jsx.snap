// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[` HomeNoLogin > matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Home-nologin"
  >
    <h1>
       Not logged in
    </h1>
    <button
      class="Home-button"
    />
    <button
      class="Home-button"
    />
  </div>
</DocumentFragment>
`;
